_source/_posts/2024-04-30-spring-api-fga.md

```diff
@@ -0,0 +1,901 @@
+}
+```
+As the `document-api` must be configured as an OAuth2 resource server, add the following properties:
+
+```yml
+# application.yml
+okta:
+  oauth2:
+    issuer: https://<your-auth0-domain>/
+    audience: https://document-api.okta.com
+```
+
+You can find your Auth0 domain with the following Auth0 CLI command:
+
+```shell
+auth0 tenants list
+```
+
+Run the API with:
+
+```shell
+./gradlew bootRun
+```
+
+Test the API authorization with curl:
+
+```shell
+curl -i localhost:8080/document
+```
+You will get HTTP response code `401` because the request requires bearer authentication. Using Auth0 CLI, get an access token:
+
+```shell
+auth0 test token -a https://document-api.okta.com -s openid
+```
+Select any available client when prompted. You also will be prompted to open a browser window and log in with a user credential. You can sign up as a new user using an email and password or using the Google social login.
+
+With curl, send a request to the API server using a bearer access token:
+
+```shell
+ACCESS_TOKEN=<auth0-access-token>
+```
+
+```shell
+curl -i --header "Authorization: Bearer $ACCESS_TOKEN" localhost:8080/document
+```
+
+You should get a JSON response listing the available documents:
+
+```json
+[
+   {
+      "id":1,
+      "parentId":null,
+      "ownerId":"test-user",
+      "name":"planning-v0",
+      "description":"Planning doc",
+      "createdTime":"2024-05-03T11:48:35.617694826",
+      "modifiedTime":"2024-05-03T11:48:35.617694826",
+      "quotaBytesUsed":null,
+      "version":null,
+      "originalFilename":null,
+      "fileExtension":".doc"
+   },
+   {
+      "id":2,
+      "parentId":null,
+      "ownerId":"test-user",
+      "name":"image-1",
+      "description":"Some image",
+      "createdTime":"2024-05-03T11:48:35.617694826",
+      "modifiedTime":"2024-05-03T11:48:35.617694826",
+      "quotaBytesUsed":null,
+      "version":null,
+      "originalFilename":null,
+      "fileExtension":".jpg"
+   },
+   {
+      "id":3,
+      "parentId":null,
+      "ownerId":"test-user",
+      "name":"meeting-notes",
+      "description":"Some text file",
+      "createdTime":"2024-05-03T11:48:35.617694826",
+      "modifiedTime":"2024-05-03T11:48:35.617694826",
+      "quotaBytesUsed":null,
+      "version":null,
+      "originalFilename":null,
+      "fileExtension":".txt"
+   }
+]
+```
+
+## Define an authorization model
+
+At a high level, an authorization model is defined by indicating user types, object types, and relationships between them. As we are not going to deep-dive on [ReBAC](https://openfga.dev/docs/authorization-concepts#what-is-relationship-based-access-control) in this guide, you can refer to OpenFGA documentation for learning about modeling concepts. Under the [Advanced use-cases](https://openfga.dev/docs/modeling/advanced) section in the doc, there is a simplified authorization model for a [Google Drive](https://openfga.dev/docs/modeling/advanced/gdrive) application ready to test:
+
+```fga
```

Comment:
fga probably is not a supported syntax, so maybe yaml or something might look better

Reply:
Use `python` - would offer some base syntax hightlighting

Reply:
Added python syntax but it does not look good in my local

Reply:
Both YAML and Python use the `#` sign for comments, so neither will be perfect. Let's ship it like this and figure out if we can add some propper FGA model highlighting added later.